core/collectionplan.class.inc.php

```diff
@@ -40,17 +40,17 @@ public function Init(): void
 	}
 
 	/**
-	 * @return \CollectionPlan
+	 * @return static
 	 */
-	public static function GetPlan(): CollectionPlan
+	public static function GetPlan(): static
```

Comment:
This is not supported by PHP 7.x, only PHP 8.0+

One solution could be to remove the return type hint completely and keep the PHPDoc as you suggested. That's what we did in iTop 3.0 for UIBlock for example.

Reply:
That's right, will change to this:
```suggestion
	public static function GetPlan()
```

Reply:
Woops, good to know, good catch Guillaume, thanks !